diff --git a/src/3rdparty/xatlas/xatlas.cpp b/src/3rdparty/xatlas/xatlas.cpp
index 5c5c57ec..5513bbb5 100644
--- a/src/3rdparty/xatlas/xatlas.cpp
+++ b/src/3rdparty/xatlas/xatlas.cpp
@@ -436,7 +436,7 @@ static void *Realloc(void *ptr, size_t size, int /*tag*/, const char * /*file*/,
 static constexpr float kPi = 3.14159265358979323846f;
 static constexpr float kPi2 = 6.28318530717958647692f;
 static constexpr float kEpsilon = 0.0001f;
-static constexpr float kAreaEpsilon = FLT_EPSILON;
+static const float kAreaEpsilon = FLT_EPSILON;
 static constexpr float kNormalEpsilon = 0.001f;
 
 static int align(int x, int a)
